Firestore/Source/API/FIRFirestoreSource.mm

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FIRFirestoreSource.h"
+
+#include "Firestore/core/src/firebase/firestore/api/source.h"
+#include "Firestore/core/src/firebase/firestore/util/hard_assert.h"
+
+namespace firebase {
+namespace firestore {
+namespace api {
+
+Source MakeSource(FIRFirestoreSource source) {
+  switch (source) {
+    case FIRFirestoreSourceDefault:
+      return Source::Default;
+    case FIRFirestoreSourceServer:
+      return Source::Server;
+    case FIRFirestoreSourceCache:
+      return Source::Cache;
+  }
+
+  UNREACHABLE();
```

Comment:
Looks like this is an alias for abort(), which we don't normally use directly from iOS (normally we'd throw an exception). Are we sure we'll get full stack trace, etc. so that if this is ever hit it will be actionable?

Reply:
We could do something more interesting, but in practice I'm not sure it's worth it.

This exists because we have exhaustive switch checking enabled. Under clang when building for iOS/macOS, we'll see warnings about uncovered cases. Even without `UNREACHABLE()` clang accepts that all paths are covered.

When building for Linux with gcc, it doesn't understand this and considers that there might be other possible values so it thinks that it's possible for there to be no match and thus you might fall out of the switch statement and then return no value from a function declared to return something.

Given that this line exists to silence a warning on Linux, changing its behavior on iOS/macOS seems like overkill. On those platforms the compiler already does the magic that prevents our getting here.